src/dialog-user/services/dialogData.ts

```diff
@@ -25,11 +25,9 @@ export default class DialogDataService {
         if (option[0] === String(field.default_value)) {
           field.selected = option;
         }
-        if (field.data_type === 'integer') {
-          dropDownValues.push([parseInt(option[0], 10), option[1]]);
-        } else {
-          dropDownValues.push(option);
-        }
+        const value = (field.data_type === 'integer' ? parseInt(option[0], 10) : option[0]);
+        const description = (parseInt(option[1], 10) ? parseInt(option[1], 10) : option[1]);
```

Comment:
Is this `description` used only for sorting?

Because if not, this breaks on any description which starts with a number.  `"3 horses"` will become just `"3"`

Reply:
Description is only used for sorting the drop down list.

Reply:
Oh.. and maybe an `_.isNaN`? 

Otherwise 0 will still get sorted last..

Reply:
ha, you beat me to it.  I just added isNan in just now.